### ISSUE TITLE
cs (capstone) and emu (unicorn) commands don't provide valid results after switching to ARM THUMB mode

### ISSUE THREAD

**Issue by mkopec3** [hugsy/gef#302]
Your issue will be closed unless you confirm the following:
- [x] Did you use the latest version of GEF from master branch?
- [x] Did you read the documentation first?
- [x] Did you check closed issues?

### Step 1: Describe your environment

  * Operating System: Linux arm-v6l-pi 4.1.19 #1 PREEMPT Tue Mar 15 18:12:17 CDT 2016 armv6l GNU/Linux
  * Architecture: armv6l
  * GDB version (including the Python library version): GDB 8.1 using Python engine 3.6

#### Steps to reproduce

The following code can be used to reproduce the issue. Check the comments for the steps required to reproduce.

```
// thumb_demo.s
// Have to be built and ran on an ARM platform
// as thumb_demo.s -o thumb_demo.o
// ld thumb_demo.o -o thumb_demo
// gdb -q ./thumb_demo
// b _start
// r
// n 2
// cs
// emu -n 1

.section .text
.global _start

_start:
        .code 32
        add r3, pc, #1	// run cs & emu -n 1, that works fine
        bx  r3		// entering into ARM THUMB mode

        .code 16
        mov r0, #1	// run cs & emu -n 1, cs & emu DONT'T PROVIDE VALID results in THUMB mode
        mov r1, #2
        add r2, r0, r1

```

#### Observed Results

  * cs (capstone-dissasembler) and emu (unicorn-emulator) commands seem to don't recognize that ARM CPU switched into THUMB mode, therefore producing no or wrong results

#### Expected results

  * GEF should identify that ARM switched into THUMB mode (based on the value in CPSR register), and then appropriately call capstone/unicorn interfaces (providing valid arch+mode values)

#### Traces

![gef_arm_thumb](https://user-images.githubusercontent.com/25580985/44203429-26c91200-a14f-11e8-893a-90ef3d689835.png)

If the above data are not enough to reproduce the issue, please let me know, so I will add some extra proofs

**Comment by hugsy** [comment]
Bug confirmed.

**Comment by hugsy** [comment]
Fixed by 4b0e110d5286fd8385df00f5781a85335d4ecced (in `dev` until 2018.10 is released - soon).
